hls4ml/writer/vivado_accelerator_writer.py

```diff
@@ -0,0 +1,357 @@
+        newfile = '{}/{}_bridge_wrapper.cpp'.format(model.config.get_output_dir(), model.config.get_project_name())
+
+        f = open(oldfile, 'r')
+        fout = open(newfile, 'w')
+
+        inp = model.get_input_variables()[0]
+        out = model.get_output_variables()[0]
+
+        for line in f.readlines():
+            if '{}.h'.format(model.config.get_project_name()) in line:
+                newline = line.replace('{}.h'.format(model.config.get_project_name()),
+                                       '{}_axi.h'.format(model.config.get_project_name()))
+            elif self.variable_definition_cpp(model, inp, name_suffix='_ap') in line:
+                newline = line.replace(self.variable_definition_cpp(model, inp, name_suffix='_ap'),
+                                       'input_axi_t {}_ap[N_IN]'.format(inp.cppname))
+            elif self.variable_definition_cpp(model, out, name_suffix='_ap') in line:
+                newline = line.replace(self.variable_definition_cpp(model, out, name_suffix='_ap'),
+                                       'output_axi_t {}_ap[N_OUT]'.format(out.cppname))
+            elif '{}('.format(model.config.get_project_name()) in line:
+                indent_amount = line.split(model.config.get_project_name())[0]
+                newline = indent_amount + '{}_axi({}_ap,{}_ap);\n'.format(model.config.get_project_name(), inp.cppname,
+                                                                          out.cppname)
+            elif inp.size_cpp() in line or inp.cppname in line or inp.type.name in line:
+                newline = line.replace(inp.size_cpp(), 'N_IN').replace(inp.type.name, 'input_axi_t')
+            elif out.size_cpp() in line or out.cppname in line or out.type.name in line:
+                newline = line.replace(out.size_cpp(), 'N_OUT').replace(out.type.name, 'output_axi_t')
+            else:
+                newline = line
+            fout.write(newline)
+
+        f.close()
+        fout.close()
+        # os.rename(newfile, oldfile)
+
+    def write_board_script(self, model):
+        '''
+        Write the tcl scripts to create a Vivado IPI project for the VivadoAccelerator
+        '''
+        filedir = os.path.dirname(os.path.abspath(__file__))
+        if self.vivado_accelerator_config.get_interface() == 'axi_stream':
+            copyfile(os.path.join(filedir, '../templates/vivado_accelerator/' +
+                                  self.vivado_accelerator_config.get_device() + '/tcl_scripts/axi_stream_design.tcl'),
+                     '{}/design.tcl'.format(model.config.get_output_dir()))
+        else: 
+            copyfile(os.path.join(filedir, '../templates/vivado_accelerator/' +
+                                  self.vivado_accelerator_config.get_device() + '/tcl_scripts/axi_lite_design.tcl'),
+                     '{}/design.tcl'.format(model.config.get_output_dir()))
```

Comment:
Shouldn't the filename of the `.tcl` script be derived somehow from the contents of the `supported_devices.json`?
And also with a check whether that interface is supported for that device against the contents of the file.